File: csharp/0344-reverse-string.cs
// While Loop Solution
class Solution
{
    public void ReverseString(char[] s)
    {
        int leftPointer = 0;
        int rightPointer = s.Length - 1;
        while (leftPointer < rightPointer)
        {
            char temp = s[leftPointer];
            s[leftPointer++] = s[rightPointer];
            s[rightPointer--] = temp;
        }
    }
}

// For Loop Solution
public class Solution
{
    public void ReverseString(char[] s)
    {
        var h = s.Length / 2;
        for (int i = 0; i < h; i++)
        {
            var temp = s[i];
            s[i] = s[s.Length - i - 1];
            s[s.Length - i - 1] = temp;
        }
    }
}

File: kotlin/0344-reverse-string.kt
/*
* Using 2 variables for the pointers
*/
class Solution {
    fun reverseString(s: CharArray): Unit {
        var low = 0
        var high = s.size-1
        while(low < high){
            val temp = s[high]
            s[high] = s[low]
            s[low] = temp
            low++
            high--
        }
    }
}

/*
* Using one variable for the pointers
*/
class Solution {
    fun reverseString(s: CharArray): Unit {
        val size = s.size
        for(i in 0 until size/2){
            val temp = s[i]
            s[i] = s[size-1-i]
            s[size-1-i] = temp
        }
    }
}
